FifthLesson/VectorProgram/VectorProgram/ProgramRun.cs

```diff
@@ -1,99 +1,118 @@
-            Vector vectorSecond = EnterUserData("second");
+            _view = view;
+            _userInput = new ProcessingUserInput(_view);
+
+            var vectors = CreateVectors();
+
+            _formattedOutput.ShowVectors(vectors[0], vectors[1]);
 
-            ShowVectorsAndActions(vectorFirst, vectorSecond);
-            ShowMultiplicationAndVectors(vectorFirst, vectorSecond);
-            CompareVectors(vectorFirst, vectorSecond);
+            CallActionsWithVectors(vectors);
 
-            Console.WriteLine($"The angle between {vectorFirst.ToString()} and {vectorSecond.ToString()} = {(vectorFirst.AngleBetweenVectors(vectorSecond)).ToString()}");
+            CallCompareVectors(vectors);
+            
+            CallAngleBetweenVectors(vectors);
 
-            Console.ReadKey();
+            _view.PressKeyToContinue();
         }
 
-        private static void CompareVectors(Vector vectorFirst, Vector vectorSecond)
+        private Vector[] CreateVectors()
         {
-            Console.WriteLine($"{vectorFirst.ToString()} == {vectorSecond.ToString()} = {(vectorFirst == vectorSecond).ToString()}");
-            Console.WriteLine($"{vectorFirst.ToString()} != {vectorSecond.ToString()} = {(vectorFirst != vectorSecond).ToString()}");
-        }
+            Vector vectorFirst = RequestVector("first");
+            Vector vectorSecond = RequestVector("second");
 
-        private void ShowMultiplicationAndVectors(Vector vectorFirst, Vector vectorSecond)
+            return new[] { vectorFirst, vectorSecond };
+        }
+        
+        private void CallActionsWithVectors(Vector[] vectors)
         {
-            Console.WriteLine("Enter a number to produce a vector by a number:");
-            var numericInput = ReadInputDoubleAndVerificate();
-            Console.WriteLine($"{vectorFirst.ToString()} * {numericInput.ToString()} = {(vectorFirst * numericInput).ToString()}");
-            Console.WriteLine($"{vectorSecond.ToString()} * {numericInput.ToString()} = {(vectorSecond * numericInput).ToString()}");
+            CallSimpleActionsWithVectors(vectors);
+            CallVectorMultiplication(vectors);
+            CallMultiplyVectorsByNumber(vectors);
+        }
 
-            Console.WriteLine($"{numericInput.ToString()} * {vectorFirst.ToString()} = {(numericInput * vectorFirst).ToString()}");
-            Console.WriteLine($"{numericInput.ToString()} * {vectorSecond.ToString()} = {(numericInput * vectorSecond).ToString()}");
+        private void CallCompareVectors(Vector[] vectors)
+        {
+            var equalityResult = (vectors[0] == vectors[1]);
+            var inequalityResult = vectors[0] != vectors[1];
+            _formattedOutput.ShowCompareVectorsResults(vectors, equalityResult, inequalityResult);
         }
 
-        private static void ShowVectorsAndActions(Vector vectorFirst, Vector vectorSecond)
+        private void CallAngleBetweenVectors(Vector[] vectors)
```

Comment:
What for do you use _Array_ as a parameter type? Why don't you check that there were exactly two vectors sent?

Reply:
It is some kind of hidden knowledge.